### PR TITLE
update expected results of slow tests

### PR DESCRIPTION
**Pull request by kashif**
Updated the two failing slow tests

### CODE REVIEW

**Review by anton-l — APPROVED** [review]
Thanks for the fixes @kashif!

@patrickvonplaten IIRC you wanted to double-check something before updating the VE tests?

**Comment by kashif** [comment]
I can run the tests now on a GPU to double check as well... will report soon

**Comment by anton-l** [comment]
Hmm, looks like the `.sum()` test is still flaky depending on the GPU, maybe that test should use tensor values instead.

**Comment by patrickvonplaten** [comment]
Ah yeah lemme run the pipeline real quick to make sure this all still works

File: tests/test_scheduler.py
Comment (patrickvonplaten) [inline tests/test_scheduler.py:717]: Circle CI seemed to be happy with `14379591680.0`: https://github.com/huggingface/diffusers/runs/8068605613?check_suite_focus=true#step:6:71 So not sure about this change

=> maybe we should increase the tolerance here?

Comment (kashif) [reply tests/test_scheduler.py:717]: yes perhaps that is what i was thinking that too so perhaps abs < 2.5 e3 or so

File: tests/test_scheduler.py
Comment (patrickvonplaten) [inline tests/test_scheduler.py:717]: Can we add a comment here that says that the results are flaky depending on GPU and this is why we have such high values?

File: tests/test_scheduler.py
Comment (patrickvonplaten) [inline tests/test_scheduler.py:285]: Can we add a comment here that says that the results are flaky depending on GPU and this is why we have such high values?

**Comment by patrickvonplaten** [comment]
@kashif, just noticed that most of our tests don't run on GPU - here a fix: https://github.com/huggingface/diffusers/pull/269

Could you maybe rebase your PR to main and check the values again? :sweat_smile:

File: tests/test_pipelines.py
Comment (patrickvonplaten) [inline tests/test_pipelines.py:301]: checking this at the moment with original code base at the moment

Comment (natolambert) [reply tests/test_pipelines.py:313]: Yeah this one I would be careful with, historically very weird and can vary across platform.

Comment (kashif) [reply tests/test_pipelines.py:313]: i tested now on osx (cpu) and linux amd64 (cpu/gpu) let me run some more tests on another server

Comment (kashif) [reply tests/test_pipelines.py:313]: yes, I believe this change doesn't look like it's due to different archs. etc. but rather some potential (hopefully) fix that caused this more significant change...

**Comment by kashif** [comment]
@natolambert @patrickvonplaten I tested now on a bunch of servers and laptop and all tests (slow ones) are passing for me...

File: tests/test_pipelines.py
Comment (patrickvonplaten) [reply tests/test_pipelines.py:313]: There is something weird here - don't get the same results anymore with our VE and the original code. Sorry give me one more day, need to go a bit deeper into the original code

Comment (kashif) [reply tests/test_pipelines.py:313]: sure yes I suspect there is something! Please take your time!

Comment (patrickvonplaten) [reply tests/test_pipelines.py:313]: Thanks!

Comment (patrickvonplaten) [reply tests/test_pipelines.py:313]: Ok double-checked it with the original repo (see https://github.com/huggingface/diffusers/pull/291). The new codebase got different results only because a different random function is used (`torch.randn(...)` instead of `torch.randnlike(...)`) => So the PR is all good for me :-)

Also ran this script a couple of times:
```python
#!/usr/bin/env python3
from diffusers import DiffusionPipeline
import torch


score_sde_sv = DiffusionPipeline.from_pretrained("google/ncsnpp-church-256")
score_sde_sv.to("cuda")

# Note this might take up to 3 minutes on a GPU
i = 33
torch.manual_seed(i)
image = score_sde_sv(num_inference_steps=1000)["sample"][0]

# save image
image.save(f"/home/patrick/images/ported_church_nsaapp_{i}_compare.png")
```

and results look good

File: tests/test_scheduler.py
Comment (patrickvonplaten) [inline tests/test_scheduler.py:285]: Hmm not very happy about this. Note that they shouldn't be that much of a difference because the model doesn't do anything - it's a dummy model and not run on GPU. It'd be nice if we could revert the change here. I think what's happening here is that we don't pass a generator to `step(...)` note that DDPM is stochastic and therefore different hardware gives different results. We can solve this simply by passing a generator with `generator = torch.manual_seed(0)`

File: tests/test_scheduler.py
Comment (patrickvonplaten) [inline tests/test_scheduler.py:717]: same here - we're getting large differences because we don't pass a generator. 
Let's maybe merge this PR: https://github.com/huggingface/diffusers/pull/289 first and then pass a generator here

File: tests/test_pipelines.py
Comment (patrickvonplaten) [reply tests/test_pipelines.py:313]: There are two things we could improve here:

1. Pass a generator through the pipeline is cleaner IMO (after having merged: https://github.com/huggingface/diffusers/pull/289)
2. This test is by far the longest test of the slow tests (takes 30+ seconds on GPU). Can we remove the number of steps from 300 to just 50 ? The model is also not that important after all

**Review by patrickvonplaten — COMMENTED** [review]
It'd be nice if we make the random scheduler tests deterministic by passing the generator correctly. Regarding the pipeline test it'd be nice to reduce the number of steps to speed it up a bit :-) 

Think merging: https://github.com/huggingface/diffusers/pull/289 can unblock this PR here

File: tests/test_scheduler.py
Comment (patrickvonplaten) [reply tests/test_scheduler.py:723]: Actually upon taking a second look I think the reason is that we don't make the test deterministic with generators - both DDPM and Score-VE schedulers make use of `torch.randn(...)`

File: tests/test_scheduler.py
Comment (kashif) [reply tests/test_scheduler.py:717]: ok

File: tests/test_pipelines.py
Comment (kashif) [reply tests/test_pipelines.py:339]: sure i can do that

File: tests/test_models_unet.py
Comment (patrickvonplaten) [inline tests/test_models_unet.py:157]: `1e-3` doesn't pass anymore?

File: tests/test_pipelines.py
Comment (patrickvonplaten) [inline tests/test_pipelines.py:277]: Why change all the values here? Do the tests fail for you locally? (Do you use a GPU or CPU?)

File: tests/test_pipelines.py
Comment (patrickvonplaten) [inline tests/test_pipelines.py:349]: That's quite a difference? Do the tests currently fail for you?

File: tests/test_pipelines.py
Comment (patrickvonplaten) [inline tests/test_pipelines.py:359]: thanks!

File: tests/test_pipelines.py
Comment (patrickvonplaten) [inline tests/test_pipelines.py:485]: Why the changes here? Does it give a different result on GPU? Should we force the test to be just CPU at the moment maybe?

Comment (kashif) [reply tests/test_pipelines.py:473]: yes so it is different on CPU -vs- GPU.... I can make it CPU only?

File: tests/test_pipelines.py
Comment (kashif) [reply tests/test_pipelines.py:339]: yes on CPU vs GPU the results are different here...

File: tests/test_pipelines.py
Comment (patrickvonplaten) [reply tests/test_pipelines.py]: Yes we should indeed do this - think @anton-l  actually merged exactly this change yesterday: https://github.com/huggingface/diffusers/commit/472425098028edc6b5689986127a91ed5addd408

Sorry for the back and forth here

Comment (kashif) [reply tests/test_pipelines.py]: thanks gonna add his changes here then

**Comment by anton-l** [comment]
@kashif looks like a couple of VE tests are failing now, could you check please? https://github.com/huggingface/diffusers/runs/8158641999?check_suite_focus=true

When those are back, I think we can carefully merge this :)

**Comment by kashif** [comment]
sure! let me get it over the line

**Comment by patrickvonplaten** [comment]
Should we still try to get this one merged?

**Comment by kashif** [comment]
yes please! I believe some of the images on the hub might need updating? let me fix the conflict

File: tests/test_pipelines.py
Comment (patrickvonplaten) [inline tests/test_pipelines.py:987]: Can we reduce this here to less inference steps ? 300 takes too long and slows us down significantly when running the whole test suite

**Comment by patrickvonplaten** [comment]
Can we fix the quality checks here and then run all slow tests to be sure?

**Review by patrickvonplaten — APPROVED** [review]

**Pull request merged**
